File: src/brs/TransactionDb.java
package brs;

import brs.schema.tables.records.TransactionRecord;
import org.jooq.DSLContext;

import java.sql.ResultSet;
import java.util.List;

public interface TransactionDb {
  Transaction findTransaction(long transactionId);

  Transaction findTransactionByFullHash(String fullHash); // TODO add byte[] method

  boolean hasTransaction(long transactionId);

  boolean hasTransactionByFullHash(String fullHash); // TODO add byte[] method

  Transaction loadTransaction(TransactionRecord transactionRecord) throws BurstException.ValidationException;

  Transaction loadTransaction(DSLContext ctx, ResultSet rs) throws BurstException.ValidationException;

  List<Transaction> findBlockTransactions(long blockId);

  void saveTransactions(List<Transaction> transactions);

}

File: src/main/java/brs/db/TransactionDb.kt
package brs.db

import brs.Transaction
import brs.schema.tables.records.TransactionRecord

interface TransactionDb : Table {
    suspend fun findTransaction(transactionId: Long): Transaction?

    suspend fun findTransactionByFullHash(fullHash: ByteArray): Transaction?

    suspend fun hasTransaction(transactionId: Long): Boolean

    suspend fun hasTransactionByFullHash(fullHash: ByteArray): Boolean

    suspend fun findBlockTransactions(blockId: Long): Collection<Transaction>

    suspend fun saveTransactions(transactions: Collection<Transaction>)
    fun loadTransaction(tr: TransactionRecord): Transaction
}
